app/models/filter.rb

```diff
@@ -140,9 +141,9 @@ def search_condition
     searches.join(' and ')
   end
 
-  def expire_topbar_cache(sweeper)
-    role.users.each      { |u| u.expire_topbar_cache(sweeper) }
-    role.usergroups.each { |g| g.expire_topbar_cache(sweeper) }
+  def expire_topbar_cache
+    role.users.each      { |u| u.expire_topbar_cache }
+    role.usergroups.each { |g| g.expire_topbar_cache }
```

Comment:
If the association has already been loaded then `find_each` will produce _additional_ database queries. I'm not sure this is a clear cut optimisation.